src/Visualization/Python/PlotSlice.py

```diff
@@ -0,0 +1,250 @@
+        "as linestyle cycles, linewidth, fontsize, legend, etc. "
+        "The stylesheet can also be set with the 'SPECTRE_MPL_STYLESHEET' "
+        "environment variable."
+    ),
+)
+def plot_slice_command(
+    h5_files,
+    subfile_name,
+    obs_id,
+    obs_time,
+    vars,
+    output,
+    slice_origin,
+    slice_radius,
+    slice_normal,
+    slice_up,
+    num_samples,
+    num_threads,
+    stylesheet,
+):
+    """Plot variables on a slice through volume data
+
+    Interpolates the volume data in the H5_FILES to a slice and plots the
+    selected variables. You choose the slice by specifying three corners of a
+    rectangle.
+    """
+    if obs_id is None:
+        raise click.UsageError("Specify '--step' / '-s' or '--time' / '-t'.")
+    if len(vars) > 1:
+        raise click.UsageError("Only one variable can be plotted at a time.")
+    var_name = vars[0]
+
+    # Interpolate the selected variable to the slice
+    if (
+        slice_origin is None
+        or slice_radius is None
+        or slice_normal is None
+        or slice_up is None
+    ):
+        raise click.UsageError(
+            f"Specify all slice parameters: '--slice-[origin,radius,normal,up]'"
+        )
+    target_coords = points_on_slice(
+        slice_origin, slice_radius, slice_normal, slice_up, num_samples
+    )
+    dim = target_coords.shape[0]
+    assert dim == 3, "Only 3D slices are supported"
+    data = np.array(
+        interpolate_to_points(
+            h5_files,
+            subfile_name=subfile_name,
+            observation_id=obs_id,
+            tensor_components=vars,
+            target_points=target_coords.reshape(3, num_samples**2),
+            num_threads=num_threads,
+        )[0]
+    ).reshape(num_samples, num_samples)
+
+    # Set axes for plot. For plotting along an axis, use the axis coordinate.
+    # Otherwise, use the affine parameter.
+    def coord_axis(vec):
+        nonzero_entries = np.nonzero(vec)[0]
+        if len(nonzero_entries) == 1:
+            return nonzero_entries[0]
+        return None
+
+    y_axis = coord_axis(slice_up)
+    x_axis = coord_axis(np.cross(slice_up, slice_normal))
+    if x_axis is not None:
+        x = target_coords[x_axis]
+        x_label = "xyz"[x_axis]
+    else:
+        x = np.linspace(0, 1, num_samples)
+        x_label = None
+    if y_axis is not None:
+        y = target_coords[y_axis]
+        y_label = "xyz"[y_axis]
+    else:
+        y = np.linspace(0, 1, num_samples)
+        y_label = None
+
+    # Apply stylesheet
+    if stylesheet is not None:
+        plt.style.use(stylesheet)
+
+    # Plot the slice
+    plt.contourf(x, y, data)
+    contours = plt.contour(
+        x, y, data, colors="white", linewidths=0.5, alpha=0.6
+    )
+    plt.clabel(contours)
+    plt.gca().set_aspect("equal")
+    plt.xlabel(x_label)
+    plt.ylabel(y_label)
+
+    if output:
+        plt.savefig(output, format="pdf", bbox_inches="tight")
```

Comment:
Check `output` for the `.pdf` extension like the other CLI plotting tools. Better yet, maybe another thing to factor out

Reply:
Ping

Reply:
Pushed a fixup that factors these commands out. Helps a lot making them consistent with each other. Note that the format and extension don't need to be specified. If no extension is specified, matplotlib defaults to png. If an extension is specified, matplotlib will detect that.
Note also that the fixup removes the automatic object label suffix from the memory monitor output file. I think it's generally better to write to the exact file that the user has specified, not modify the name unexpectedly. This makes the output easier to find and, more importantly, avoids accidentally overwriting existing files. Also simplifies the code.

Reply:
I think you mean the `PlotSizeControl` test? But ok yeah that makes sense, just be sure to edit the test